In [23]:
!apt install git
!git config --global user.name "somineda"
!git config --global user.email "fuurele@gmail.com"


Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
git is already the newest version (1:2.34.1-1ubuntu1.12).
0 upgraded, 0 newly installed, 0 to remove and 29 not upgraded.


안녕하세요.<br><br>
수강생 여러분~! Chapter04 수업은 잘 들으셨나요?

Chapter04의 수업 내용을 바탕으로 아래 문제를 풀어주시면 됩니다.

<br>

[**답변 입력 방법**]

문제 아래에 있는 코드 입력창에 정답코드를 작성하신 후  __▷ 버튼을 클릭해 코드를 실행__ 해보세요<br>
<br>

-----

<br>

__[문제0]__

한 변의 길이가 5인 정사각형이 있습니다.

정사각형의 넓이를 구하세요.
<br>

In [1]:
print(5*5)

25


<br>

__[문제1]__

오즈 코딩 스쿨 백엔드 8기 과정에 참여한 수강생 중 5명의 수강생이 자율적으로 스터디 그룹을 결성했습니다.

스터디 그룹 내에는 한 주간 배운 내용으로 매주 쪽지 시험을 보는 룰이 있습니다.

이 쪽지 시험의 평균이 70점이 넘으면 주말 스터디는 없습니다.

이들의 평균 점수를 대신 계산해주세요.

<br>
A: 65, B: 75, C: 50, D: 80, E: 60

<br>

In [2]:
A = 65
B = 75
C = 50
D = 80
E = 60

avg_score = (A + B + C + D + E) / 5
print(avg_score)


66.0


<br>__[문제2]__

 **키 : 180cm , 몸무게 92kg**

__Tips__


 **신체질량지수(BMI)를 통해 비만도를 대략적으로 파악해 볼 수 있습니다.<br>
 첨부된 그림에 있는 계산식을 파이썬 코드로 작성해보세요**


- 단위를 고려하세요 !

In [10]:
kg = 92
height = 180 / 100
bmi = kg / (height ** 2)
print(bmi)


28.39506172839506


<br>__[문제3]__

밑변과 높이가 주어졌을때 피타고라스 정의를 이용해 빗면을 구할 수 있습니다.

아래 주어진 밑변과 높이를 이용해 빗면을 구해주세요



__Tips__


- 피타고라스 정의

높이(a) : 13cm, 밑변(b) : 9cm,  빗면(c) : ?

In [11]:
a = 13
b = 9

c = (a ** 2 + b ** 2) ** 0.5

print(c)

15.811388300841896


__[문제4]__

쉬지도 않고 일하는 직장인 오즈씨는 3년만에 첫 성과급으로 2500만원을 받았습니다.

연이율 6.15%의 고정금리 상품에 예금하려고 합니다.

3년 동안 넣어두면 원금을 제외한 이자가 얼마나 되는지 계산해 주세요.


<br>

__Tips__


- 이자 지급 방식은 매년 원금 기준으로 지급되는 단리 방식입니다. 단리 방식 계산식을 검색해보세요!)

원금 25000000원, 연이율 6.15%, 기간 : 3년

In [12]:
salary = 25000000
rate = 6.15 / 100
years = 3


interest = salary * rate * years

print("이자:" , interest)



이자: 4612500.000000001


## 도전! 신비의 섬 탐험

### [배경 이야기]

탐험가인 당신은 신비의 섬에 도착했습니다.


이 섬에는 숨겨진 보물이 있으며, 이를 찾기 위해 여러 도전 과제를 해결해야 합니다.


섬에는 알 수 없는 외계인이 당신을 쫓아오고 있어 모든 도전을 신속하게 해결하고 섬을 탈출해야 합니다.


<br>

### [문제 풀이 팁]

신비의 섬을 나가기 위해 필요한 비밀번호를 변수에 담아 놓는다면 번거로운 계산 과정이 줄어듭니다.


- 아직 변수에 대해 배운 적이 없다면?


__Chapter02 강의 예습 혹은 검색을 통해 간단한 개념을 파악할 수 있습니다!__


<br>

##### __<신비의 섬 탐험> 문제는 성장을 위해 난이도가 살짝 높게 작성되었기 때문에 어려울 수 있지만 도전해보시길 바랍니다 :)__



<br>

__[문제 1: 입구의 수수께끼]__

<br>

섬의 입구에 도착한 당신은 좋지 않은 느낌을 받습니다. 무언가 비릿한 냄새가 여기저기에서 납니다.

아마도 나보다 앞서 도착한 도전자들의 시체가 썩는 냄새인 것 같습니다.

지금...누군가 당신을 지켜보고 있습니다. 한시라도 빠르게 이 섬을 탈출해야 합니다.

우선 이 거대한 돌문을 열고 섬의 더 깊숙한 곳으로 들어가야 합니다. 돌문을 열려면 두 개의 열쇠 숫자를 더해야 합니다.

열쇠 숫자는 14와 29입니다.

돌문을 여는 숫자를 구해주세요.

In [13]:
key1 = 14
key2 = 29

rockdoor = key1 + key2

print("돌문을 여는 숫자는", rockdoor, "입니다.")

돌문을 여는 숫자는 43 입니다.


<br>

__[문제 2: 첫 번째 강의 다리]__

<br>



돌문을 지나자 거친 강물이 흐르는 강이 나타났습니다.

아주 낡은 나무 다리가 힘겹게 메달려 있습니다.

당장이라도 무너질 것 같지만 이 다리 말고는 반대편으로 갈 수 있는 길이 보이지 않습니다.

무언가 불길한 기운이 계속 가까워지는 것만 같아 당신 빠르게 다리를 건너야 합니다.

다리는 50kg 이상 버티지 못하기 때문에 68kg인 당신은 무게를 줄여야 합니다.

방금 전 돌문을 통과하면서 얻은 마법 룬을 이용해 무게를 줄일 수 있습니다.

다리를 건너기 위해 줄여야 할 최소 무게를 구해주세요. (이상, 이하, 초과, 미만은 신경쓰지 않습니다!)


<br>

In [14]:
current_weight = 68
limit = 50
weight = current_weight - limit


print(weight)


18




<br>

__[문제 3: 숲 속의 고목나무]__

<br>



첫 번째 강을 건너자 으스스한 숲 속에 들어섰습니다.

귀를 찢는 울음소리가 숲 속에 울려 퍼집니다. 당신은 바로 오래된 고목나무 숲으로 몸을 숨깁니다.

숲 깊이 들어가자 고목 나무에 걸린 현판이 보입니다.

"나무의 나이를 구하고 연못으로 가라"

나무의 나이를 구하기 위해서는 4년마다 한 번의 나이테가 생깁니다.

돌문을 열때 사용한 숫자를 나이테 개수로 사용하여 나무의 나이를 구하세요.



__Tips__
- 문제 1의 결과값을 참고하세요 !

In [15]:
tree_age = rockdoor * 4
print(tree_age)

172


<br>

__[문제 4: 숲 속의 마법 연못]__

<br>



고목나무를 지나자 신비로운 마법 연못이 나타났습니다.

연못 속에는 불가사의한 에너지가 흐르고 있습니다.

이 에너지를 이용하면 보물을 찾는 데 필요한 단서를 얻을 수 있습니다.

연못의 깊이를 측정하려면 나무의 나이를 이용해야 합니다.

나무의 나이가 172살이고, 연못의 깊이는 나무 나이의 1/4입니다.

연못의 깊이를 구하세요.

<br>



In [16]:
pond = tree_age  / 4

print(pond)

43.0


<br>

__[문제 5: 사원의 입구]__

<br>



연못을 지나니 고대 사원이 나타났습니다. 사원의 입구에는 비밀번호를 입력해야 합니다.

비밀번호는 3자리 수로, 첫 번째 자리는 5, 두 번째 자리는 돌문을 여는 숫자의 십의 자리 수, 세 번째 자리는 돌문을 여는 숫자의 일의 자리 수의 3분의 1입니다.

비밀번호를 구하세요.

<br>

In [17]:
num1 = 5
num2 = rockdoor // 10
num3 = rockdoor % 10 //3

pw = num1 * 100 + num2 * 10 + num3

print(pw)

541


<br>

__[문제 6: 비밀의 동굴]__

<br>

사원 내부에 비밀의 동굴로 이어지는 통로가 있습니다.

무언가 쿵 하고 넘어지는 소리가 들렸습니다. 신경을 곤두세우는 까마귀 소리가 여기저기서 울려 퍼집니다.

주변을 멤돌던 존재가 사원에 진입한것 같습니다.

어서 동굴을 벗어나야할 것 같습니다.

동굴을 열려면 암호를 풀어야 합니다. 암호는 돌문을 열때 사용한 숫자의 제곱근과 같습니다.

암호를 구하세요.

<br>

In [18]:
dong_pw = rockdoor ** 0.5
print(dong_pw)

6.557438524302


<br>

__[문제 7: 해변의 파도]__

<br>


비밀의 동굴을 지나 해변에 도착했습니다.

동굴에서 빠져나오진 못한 존재는 엄청난 촉수를 가진 거대 외계인으로 보입니다. 외계인은 당신을 보며 군침을 흘리고 있습니다. 빠르게 이곳을 탈출해 섬의 반대편으로 도망가야 합니다.

배를 타고 빠르게 벗어나기 위해 우선 파도를 파악해야합니다.

파도가 일정한 간격으로 오고 있음을 발견했습니다.

5분 동안 20번의 파도가 왔다면, 파도 한 번이 오는 데 걸리는 평균 시간을 구하세요.

<br>

In [19]:
time = 5
pado = 20
average_wave = time / pado

average_wave


0.25

<br>

__[문제 8: 폭포의 높이]__

<br>

섬의 반대편의 무사히 도착했습니다.

거대한 폭포가 당신의 길을 가로 막고 있습니다.

자세히 보니 폭포 위쪽에 반짝이는 무언가가 있습니다. 당신이 찾던 보물일 수 있습니다.

마법 룬을 이용해 폭포의 높이를 구하고 폭포의 위로 올라갑니다.

물방울이 떨어져 땅에 닿는 데 걸리는 시간은 2초입니다. 중력가속도는 9.8m/s²입니다.

돌문을 열때 사용한 숫자를 이용해 폭포의 높이를 구하세요. (단, 공기저항은 무시합니다.)

<br>


__Tips__
- 높이 = 0.5 * 중력가속도 * (시간 + 돌문의 숫자의 마지막 자리 수)²

<br>

In [20]:
height1 = 0.5 * 9.8 * (2 + rockdoor % 10) **2

print(height1)

122.50000000000001


<br>

__[최종 문제: 탈출 비밀번호]__

<br>

보물을 찾았습니다.






쿠우에에이이이이이이


에일리언이 당신을 쫓아왔습니다. 당신은 곧 에일리언의 좋은 영양소가 될 것 같습니다.

당신은 지금 정신을 차리고 앞에 보이는 에일리언의 우주선을 뻇어 섬을 탈출해야합니다.

우주선에 타기 위해서는 비밀번호를 입력해야 합니다.

비밀번호는 현재 좌표의 첫 번째 숫자와 세 번째 숫자의 차와 같습니다.

비밀번호를 알아내세요!

<br>

현재 좌표 : 32, 43, 128

<br>

In [21]:
current1 = 32
current3 = 128

pwd = current1 - current3

print(pwd)

-96


In [37]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [41]:
%cd /content/python


/content/python


In [42]:
import shutil

# 파일 경로
colab_file = "/content/drive/MyDrive/Ch04_task(code).ipynb"
github_repo_path = "/content/python/Ch04_task(code).ipynb"

# 파일 복사 (원본은 유지)
shutil.copy(colab_file, github_repo_path)


'/content/python/Ch04_task(code).ipynb'

In [43]:
!ls /content/python


'Ch04_task(code).ipynb'


In [45]:
%cd /content
!git clone https://github.com/somineda/python.git


/content
fatal: destination path 'python' already exists and is not an empty directory.
